kbfscrypto/encrypted_data.go

```diff
@@ -183,12 +214,50 @@ func EncryptPaddedEncodedBlock(paddedEncodedBlock []byte, key BlockCryptKey) (
 	return EncryptedBlock{encryptedData}, nil
 }
 
+// EncryptPaddedEncodedBlockV2 encrypts a padded, encoded block, using
+// the v2 block encryption scheme.
+func EncryptPaddedEncodedBlockV2(paddedEncodedBlock []byte, key BlockHashKey) (
```

Comment:
For every V2 function you add that has a V1 version, what do you think of renaming the existing one to have a V1 suffix, and/or having an unsuffixed function that switches on the version and delegates to the right one? Not sure if you're gonna do this in future PRs...

Reply:
I thought about having an unsuffixed one, but since the functions require different key types, that seemed too unwieldy.  (That is, would we have to make and pass both key types into the function, regardless of the type in use?)  I guess the alternative would be opening up the interfaces to take the TLF crypt key, the server side key half, and the desired version all at once...

Using V1 instead of V2 makes sense, though maybe we shouldn't switch it until v2 is the default?  i.e., `kbfshash.DefaultHash` should probably remain as it is for now, with an explicit call to v2 when needed.  If you have strong feelings here, I'm definitely open to changing it now though.

Reply:
Yeah, I think having the general function take in the TLF crypt key, etc. was what I was thinking, if that also works for all the other V1/V2 functions.

I don't think `DefaultHash` should change, just these V1/V2 functions. I don't have strong feelings on when to rename, although if you end up having the general function taking a version in `kbfscrypto`, that kinda forces a rename.

Reply:
Ok.  I'll try to turn it into a general function, but unless you prefer otherwise, I'm going to do it in a new cleanup PR ordered after #1837 and #1838, since those are so dependent on this function scheme, it'll be a mess to try to change them.